panel/lxqtpanel.cpp

```diff
@@ -221,15 +225,39 @@ LXQtPanel::LXQtPanel(const QString &configGroup, LXQt::Settings *settings, QWidg
     connect(LXQt::Settings::globalSettings(), &LXQt::GlobalSettings::settingsChanged, this, [this] { update(); } );
     connect(lxqtApp,                          &LXQt::Application::themeChanged,       this, &LXQtPanel::realign);
 
-    connect(mStandaloneWindows.data(), &WindowNotifier::firstShown, this, [this] { showPanel(true); });
-    connect(mStandaloneWindows.data(), &WindowNotifier::lastHidden, this, &LXQtPanel::hidePanel);
+    connect(mStandaloneWindows.get(), &WindowNotifier::firstShown, this, [this] { showPanel(true); });
+    connect(mStandaloneWindows.get(), &WindowNotifier::lastHidden, this, &LXQtPanel::hidePanel);
 
     readSettings();
 
     ensureVisible();
 
     loadPlugins();
 
+    if(qGuiApp->nativeInterface<QNativeInterface::QWaylandApplication>())
+    {
+        // Create backing QWindow for LayerShellQt integration
+        create();
+
+        // Init Layer Shell (Must be done before showing widget)
+        mLayerWindow = LayerShellQt::Window::get(windowHandle());
+        mLayerWindow->setLayer(LayerShellQt::Window::LayerTop);
+
+        mLayerWindow->setScope(QStringLiteral("dock"));
+
+        LayerShellQt::Window::Anchors anchors;
+        anchors.setFlag(LayerShellQt::Window::AnchorLeft);
+        anchors.setFlag(LayerShellQt::Window::AnchorBottom);
+        anchors.setFlag(LayerShellQt::Window::AnchorRight);
+        mLayerWindow->setAnchors(anchors);
+
+        mLayerWindow->setKeyboardInteractivity(LayerShellQt::Window::KeyboardInteractivityNone);
```

Comment:
This should be `KeyboardInteractivityOnDemand` otherwise typing in search and using arrow keys won't work in most compositors. See https://github.com/labwc/labwc/pull/1594

Reply:
Oh, and `Ondemand` fixed also those popup opening as normal windows in add-favorites dialog, directorymenu, mount and else!

Reply:
Done. Thanks for the hint.
Here starting nested labwc with your `startlxqtlabwc` script it doesn't solve the issue.
FancyMenu does not receive keyboard input. Anyway it's more correct now

Reply:
Just to be sure I've started labwc session directly from SDDM to double check.
I've also tried setting "Exclusive" keyboard focus.
Still not working.

Reply:
You need latest labwc, they fixed it today.

Reply:
> pcmanfm-qt --desktop had the same issue, no keyboard.

Yes, the keyboard focus problem of pcmanfm-qt's desktop was also fixed by a patch. LabWC's devs are very fast.